From f1652b60a9fa0580c5fc60758c3ccdfcf691b5c2 Mon Sep 17 00:00:00 2001
From: Daniel Drake <dsd@laptop.org>
Date: Mon, 17 Dec 2012 14:58:20 +0000
Subject: [PATCH] na: apply style after realize

Under Fedora 18 (GTK-3.6.2), notification icons such as the battery/power
icon from gnome-settings-daemon are appearing all-black on top of a black
background.

This is because style_updated is being called before the tray applet
is realized, meaning that the following code in
na_tray_applet_style_updated() causes a bail-out:

  if (!applet->priv->tray)
    return;

and the style is never applied.

Fix this by updating the style explicitly after realizing the widget.
---
 applets/notification_area/main.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/applets/notification_area/main.c b/applets/notification_area/main.c
index 5fc4d1d..0d5de70 100644
--- a/applets/notification_area/main.c
+++ b/applets/notification_area/main.c
@@ -36,6 +36,8 @@
 
 #define NOTIFICATION_AREA_ICON "gnome-panel-notification-area"
 
+static void na_tray_applet_style_updated (GtkWidget *widget);
+
 struct _NaTrayAppletPrivate
 {
   NaTray *tray;
@@ -79,6 +81,7 @@ na_tray_applet_realize (GtkWidget *widget)
 
   gtk_container_add (GTK_CONTAINER (widget), GTK_WIDGET (applet->priv->tray));
   gtk_widget_show (GTK_WIDGET (applet->priv->tray));
+  na_tray_applet_style_updated (widget);
 }
 
 static void
-- 
1.8.0.1

